### ISSUE TITLE
Inconsistent ID Assignment During Training on Custom Driving Dataset

### ISSUE THREAD

**Issue by back2zack** [megvii-research/MOTRv2#69]
**Description:**
I'm encountering issues with the MOTRV2 model while training it on a custom dataset of driving sequences . The model detects objects in the frames correctly, but the ID assignment is highly inconsistent. Objects often get new IDs in consecutive frames, and multiple IDS are sometimes assigned to the same Object.

**Dataset Details:**

- The dataset consists of sequences with approximately 100 frames each.
- Each sequence captures various driving scenes .
- Attempted to overfit on a single sequence with 150 frames.

**Data Preparation:**
I used the same data augmentation techniques and data structure as those used for the DanceTrack dataset.

**Training Parameters:**

`--meta_arch motr
--dataset_file e2e_dspace
--epoch 60
--with_box_refine
--lr_drop 4
--lr 2e-4
--lr_backbone 2e-5
--batch_size 1
--sample_mode random_interval
--sample_interval 4
--sampler_lengths 10
--merger_dropout 0
--dropout 0
--random_drop 0.1
--fp_ratio 0.3
--query_interaction_layer QIMv2
--query_denoise 0.05
--num_queries 10
`

**Observed Behavior:**

- Objects detected in frames are frequently assigned new IDs.
- Multiple IDs are sometimes given to the same Object.
- Although the loss function is decreasing, the ID assignment remains inconsistent.

![image](https://github.com/megvii-research/MOTRv2/assets/93681988/d78a5db0-c39b-460f-aee2-7d91e5ad5345)
![image](https://github.com/megvii-research/MOTRv2/assets/93681988/cad3fc5a-ba05-423d-aeda-64f70ddb7945)
![image](https://github.com/megvii-research/MOTRv2/assets/93681988/15c37a7d-aa67-43ab-9ea6-5cae4120eb60)

I assume that the inaccuracy in detections is due to the fact that the system was only trained on a small sequence, but there is no correct assignment of an ID during the entire sequence.

**Comment by Shawnnnnn** [comment]
what is your score threshold when inference? when I set it to 0.5, it will get probably OK results.

**Comment by back2zack** [comment]
@Shawnnnnn  are you also working with automotive dataset ?  how are your results looking ?
